botometerAnalyser/pipelines/retweets.js

```diff
@@ -0,0 +1,89 @@
+const request = require('request-promise');
+
+const usersAnalysis = require('./usersAnalysis');
+const { retweeterIdsQueue } = require('../queues/retweeters');
+const { getTweetQueue } = require('../queues/getTweet');
+
+
+async function analyse({ screenName, tweetId, tweetUrl, responseUrl, requesterUsername }) {
+	getTweetQueue.add({
+		screenName,
+		tweetId,
+		tweetUrl,
+		responseUrl,
+		requesterUsername
+	});
+}
+
+getTweetQueue.on('completed', onGetTweetCompleted);
+getTweetQueue.on('failed', failed);
+
+async function failed(job) {
+	const { tweetUrl, responseUrl, requesterUsername } = job.data;
+
+	return request({
+		url: responseUrl,
+		method: 'POST',
+		json: {
+			text: `@${requesterUsername} I could not found the tweet ${tweetUrl}. Are you sure you spelled it correctly?`,
+			response_type: 'in_channel'
+		},
+	});
+}
+
+
+async function onGetTweetCompleted(job, result) {
+	const {	screenName, responseUrl, tweetId, requesterUsername } = job.data;
+	const tweet = result.data;
+
+	retweeterIdsQueue.add({
+		screenName,
+		tweet,
+		tweetId,
+		responseUrl,
+		requesterUsername
+	});
+}
+
+
+retweeterIdsQueue.on('completed', onRetweetersCompleted);
+
+async function onRetweetersCompleted(job, result) {
+	try {
+		const { screenName, tweet, responseUrl, requesterUsername } = job.data;
+		const retweeterIds = result.data.ids;
+
+		// If there is no search results, nothing to do
+		if (!retweeterIds.length) {
+			return request({
+				url: responseUrl,
+				method: 'POST',
+				json: {
+					text: `@${requesterUsername} Nobody retweeted the tweet "${tweet.id_str}"`,
+					response_type: 'in_channel'
+				},
+			});
+		}
+
+		const users = retweeterIds.map(userId => ({ userId }));
+
+		await usersAnalysis.scheduleUsersAnalysis({
+			users,
```

Comment:
Here users is an array of ids (`[{userId}, {userId}, …]`) only whereas when we schedule jobs with a text search param we pass an array with screenName and user id (`[{screenName, userId}, …]`). We should unify this as it's bug prone. 
Moreover, now we have logs like `Botometer job for user undefined completed`. 
I think we should discuss this.

Reply:
Replaced by `[{userId}, {userId}, …]` everywhere.